service/routes.py

```diff
@@ -89,6 +90,126 @@ def create_wishlists():
+    wishlistproduct.wishlist_id = wishlist_id
+    if not Wishlist.find_by_id( wishlist_id ):
+        return(
+            f"Wishlist with id {wishlist_id} was not found",
+            status.HTTP_404_NOT_FOUND
+        )
+    wishlistproduct.create()
+    message = wishlistproduct.serialize()
+
+    return make_response(jsonify(message), status.HTTP_201_CREATED)
+
+######################################################################
+# LIST PRODUCTS IN A WISHLIST
+######################################################################
+@app.route("/wishlists/<int:wishlist_id>/products", methods=["GET"])
+def list_products_in_wishlist(wishlist_id):
+    """
+    List all products in a wishlist based on a wishlist_id
+    """
+    app.logger.info("Request to list products in a wishlist")
+    wishlist_products= WishlistProduct.find_all_by_wishlist_id(wishlist_id)
+    products_id_list = [ wishlist_product.product_id for wishlist_product in wishlist_products ]
+    res = []
+    for product_id in products_id_list:
+        product = Product.find_by_id( product_id )
+        if not product:
+            return(
+                f"Product with id {product_id} was not found",
+                status.HTTP_404_NOT_FOUND
+            )
+        res.append( product.serialize() )
+    return make_response(jsonify(res), status.HTTP_200_OK)
+
+######################################################################
+# CREATE A NEW PRODUCT
+######################################################################
+@app.route("/products", methods=["POST"])
+def create_products():
+    """
+    Creates a product
+    This endpoint will create a product based the data in the body that is posted
+    """
+    app.logger.info("Request to create a product")
+    data = request.get_json()
+    product= Product()
+    if isinstance(data["status"], str):
+        if not hasattr(Availability, data["status"]):
+            return(
+                f"product got wrong status",
+                status.HTTP_400_BAD_REQUEST
+            )
+        data["status"] = getattr(Availability, data["status"])
+    product.deserialize(data)
+    product.create()
+    return make_response(
+        jsonify( product.serialize() ),
+        status.HTTP_201_CREATED
+    )
+
+######################################################################
+# RETRIEVE A PRODUCT
+######################################################################
+@app.route("/products/<int:product_id>", methods=["GET"])
+def get_product(product_id):
+    """
+    Retrieve a single product
+    This endpoint will return a product based on it"s id
+    """
+    app.logger.info("Request for product with id: %s", product_id)
+    product = Product.find_by_id(product_id)
+    if not product:
+        return(
+            f"product with id {product_id} was not found",
+            status.HTTP_404_NOT_FOUND
+        )
+    return make_response(jsonify(product.serialize()), status.HTTP_200_OK)
```

Comment:
I really don't think this is part of our scope. Our resource is "Wishlist". And while I agree with you that there should be a way for us to add products to the database to make sure the wishlists functionality works properly, retrieving products holds no value. It doesn't hurt, but I'm wondering if the professor might deem it as our misunderstanding of the scope. We can consult with him.